File: 백준/Silver/10828. 스택/스택.py
import sys
lst = []
for i in range(int(sys.stdin.readline())):
    command = sys.stdin.readline().split()
    if command[0] == "push": lst.append(int(command[1]))
    elif command[0] == "pop": print(-1 if lst == [] else lst.pop())
    elif command[0] == "size": print(len(lst))
    elif command[0] == "empty": print(1 if lst == [] else 0)
    elif command[0] == "top": print(-1 if lst == [] else lst[-1])

File: 백준/Silver/10828. 스택/스택.java

import java.io.*;
import java.util.HashMap;
import java.util.Map;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		Stack<Integer> stack = new Stack<>();
		StringTokenizer st;
		for (int i = 0; i < N; i++) {
			st = new StringTokenizer(br.readLine());
			String command = st.nextToken();
			if ("push".equals(command)) {
				int value = Integer.parseInt(st.nextToken());
				stack.push(value);
			} else if ("pop".equals(command)) {
				if (stack.isEmpty()) {
					System.out.println(-1);
				} else {
					System.out.println(stack.pop());
				}
			} else if ("size".equals(command)) {
				System.out.println(stack.size());
			} else if ("empty".equals(command)) {
				if (stack.isEmpty()) {
					System.out.println(1);
				} else {
					System.out.println(0);
				}
			} else if ("top".equals(command)) {
				if (stack.isEmpty()) {
					System.out.println(-1);
				} else {
					System.out.println(stack.peek());
				}
			}
		}
	}
}